### ISSUE TITLE
json serialization/deserialization of keys and stores does not work

### ISSUE THREAD

**Issue by angleto** [Auties00/Whatsapp4j#242]
The deserialization of a previously serialized Keys or Store data structure raise exceptions.
The errors are like the following:
```
com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot deserialize value of type `java.util.concurrent.ConcurrentHashMap<it.auties.whatsapp.model.contact.ContactJid,java.util.concurrent.ConcurrentLinkedDeque<it.auties.whatsapp.model.info.MessageInfo>>` from Array value (token `JsonToken.START_ARRAY`)
```

To reproduce the issue just serialize complete Keys and Store classes through the `toJson` function and deserialize them back using the ObjectMapper class.

**Comment by Auties00** [comment]
> The deserialization of a previously serialized Keys or Store data structure raise exceptions. The errors are like the following:
> 
> ```
> com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot deserialize value of type `java.util.concurrent.ConcurrentHashMap<it.auties.whatsapp.model.contact.ContactJid,java.util.concurrent.ConcurrentLinkedDeque<it.auties.whatsapp.model.info.MessageInfo>>` from Array value (token `JsonToken.START_ARRAY`)
> ```
> 
> To reproduce the issue just serialize complete Keys and Store classes through the `toJson` function and deserialize them back using the ObjectMapper class.

What version are you using?

**Comment by angleto** [comment]
> 
> What version are you using?

I can reproduce the issue with the data structures from the master version.

**Comment by Auties00** [comment]
> > 
> > What version are you using?
> 
> I can reproduce the issue with the data structures from the master version.

Delete that session and create a new one. You probably changed version without doing it which is not something you can do

**Comment by angleto** [comment]
> > 
> > What version are you using?
> 
> I can reproduce the issue with the data structures from the master version.

No, that's not the case. I deleted the keys and store and recreated them, but the problem persists especially after connecting and downloading the data (contacts, conversations) from WhatsApp. Is there any unit test that verify full serialization of stores and keys? May the they would reproduce the issue.
Thanks

**Comment by angleto** [comment]
I was able to make it working using your JSON utility class: I didn't notice it before, I used toJson to serialize and an Object mapper for deserialization. Though it was necessary to enable the following properties building the ObjectMapper on the Json class:
```
                    .enable(WRITE_ENUMS_USING_INDEX)
                    .enable(READ_ENUMS_USING_TO_STRING)
```

I updated a few dependencies as well, If you find the changes useful they are in the PR #224 already opened.
